cdk/domino_cdk/provisioners/vpc.py

```diff
@@ -193,20 +189,30 @@ def provision_bastion(self, name: str, bastion: config.VPC.Bastion) -> Optional[
         if not bastion.enabled:
             return None
 
+        bastion_opts: Dict[Any, Any] = {}
+
         if bastion.ami_id:
             region = cdk.Stack.of(self.scope).region
-            machine_image = ec2.MachineImage.generic_linux(
+            bastion_opts["machine_image"] = ec2.MachineImage.generic_linux(
                 {region: bastion.ami_id},
                 user_data=ec2.UserData.custom(bastion.user_data) if bastion.user_data else None,
             )
-
-            root_device_name = root_device_mapping(self.scope, bastion.ami_id).name
         else:
-            machine_image = ec2.GenericSSMParameterImage(
+            bastion_opts["machine_image"] = ec2.GenericSSMParameterImage(
                 "/aws/service/ami-amazon-linux-latest/amzn2-ami-hvm-x86_64-gp2", ec2.OperatingSystemType.LINUX
```

Comment:
Why we are using a specific image here instead of default?

Reply:
I don't think there is a default? This is the latest Amazon Linux 2 AMI.

Reply:
I mean if we do not specify an image then latest AL2 is automatically picked. 

Or, IIRC, there is a function to pick default image by family.

Reply:
This is that function IMO?